routing/DELEGATED_CONTENT_ROUTING_HTTP.md

```diff
@@ -9,91 +9,221 @@
+- `Addrs`: a list of known multiaddrs for the peer
+  - This list may be incomplete or incorrect and should only be treated as *hints* to improve performance by avoiding extra peer lookups
+
+The server should respect a passed `transport` query parameter by filtering against the `Addrs` list.
+
+
+#### Write Provider Records
+
+```json
+{
+    "Protocol": "bitswap",
+    "Signature": "<signature>",
+    "Payload": "<payload>"
+}
+```
+
+- `Signature`: a multibase-encoded signature of the sha256 hash of the `Payload` field, signed using the private key of the Peer ID specified in the `Payload` JSON
+  - Servers may ignore this field if they do not require signature verification
+- `Payload`: a string containing a serialized JSON object which conforms with the following schema:
+```json
+{
+    "Keys": ["cid1", "cid2"],
+    "Timestamp": 0,
+    "AdvisoryTTL": 0,
+    "ID": "12D3K...",
+    "Addrs": ["/ip4/..."],
+}
+```
+  - `Keys` is a list of the CIDs being provided
+  - `Timestamp` is the current time
+  - `AdvisoryTTL` is the time by which the caller expects the server to keep the record available
+    - If this value is unknown, the caller may use a value of 0
+  - `ID` is the peer ID that was used to sign the record
+  - `Addrs` is a list of string-encoded multiaddrs
+
+A 403 response code should be returned if the signature check fails.
+
+Note that this only supports Peer IDs expressed as identity multihashes. Peer IDs with older key types that exceed 42 bytes are not verifiable since they only contain a hash of the key, not the key itself. Normally, if the Peer ID contains only a hash of the key, then the key is obtained out-of-band (e.g. by fetching the block via IPFS). If support for these Peer IDs is needed in the future, this spec can be updated to allow the client to provide the key and key type out-of-band by adding optional `PublicKey` and `PublicKeyType` fields, and if the Peer ID is a CID, then the server can verify the public key's authenticity against the CID, and then proceed with the rest of the verification scheme.
+
+The `Payload` field is a string, not a proper JSON object, to prevent its contents from being accidentally parsed and re-encoded by intermediaries, which may change the order of JSON fields and thus cause the record to fail validation.
+
+#### Write Provider Records Response
+```json
+{
+    "AdvisoryTTL": 0
+}
+```
+
+- `AdvisoryTTL` is the time at which the server expects itself to drop the record
+  - If less than the `AdvisoryTTL` in the request, then the client should re-issue the request by that point
+  - If greater than the `AdvisoryTTL` in the request, then the server expects the client to be responsible for the content for up to that amount of time (TODO: this is ambiguous)
+  - If 0, the server makes no claims about the lifetime of the record
+
+
+### filecoin-graphsync
+Multicodec code: 0x0910
+
+#### Read Provider Records
+
+```json
+{
+    "Protocol": "filecoin-graphsync",
```

Comment:
why has the '-v1' been dropped from the name? we should use the multicodec table values unless there's a strong reason to diverge